xaitk_saliency/utils/gen_coco_sal.py

```diff
@@ -0,0 +1,262 @@
+import numpy as np
+from typing import Optional, Dict, Callable, Any, Iterable, List, Hashable, Tuple, KeysView
+import logging
+from PIL import Image  # type: ignore
+
+from smqtk_detection.interfaces.detect_image_objects import DetectImageObjects
+from smqtk_detection.utils.bbox import AxisAlignedBoundingBox
+
+from xaitk_saliency import GenerateDetectorProposalSaliency, PerturbImage
+from xaitk_saliency.utils.masking import occlude_image_batch
+from xaitk_saliency.utils.detection import format_detection
+from xaitk_saliency.utils.exceptions import MismatchedLabelsError
+
+try:
+    import kwcoco  # type: ignore
+    is_usable = True
+except ModuleNotFoundError:
+    is_usable = False
+
+
+LOG = logging.getLogger(__name__)
+
+
+# mute logger to suppress verbose messages
+def _mute_log(_: Any) -> None: ...
+
+
+if not is_usable:
+    print("This tool requires additional dependencies, please install 'xaitk-saliency[tools]'")
```

Comment:
This should use `LOG.warning` or `warnings.warn` instead of a print. That level of logging should always be printed.

We should also be more specific than "this tool" and more specifically refer to the `__name__` of this module. That way the user knows where this is coming from (though I suppose `warnings.warn` would also denote this location).